### PR TITLE
Add Infernal's cmscan and HMMER's hmmsearch/hmmscan tools

### PR DESCRIPTION
**Pull request by mjmansfi**
# luslab-nf-modules

Thank you for contributing and submitting a pull request!

Please check you have done the following:

## Uniform coding style

- [x] Used snake_case rather than camelCase
- [x] Followed the output channel naming convention (e.g. `bam` for BAM files, `bed` for BED files)

## Continuous integration

- [x] Added any new modules to `.github/workflows/modules-testing.yml`
- [x] Added any new containers to `.github/workflows/docker-build-push.yml` and `.github/workflows/docker-linting.yml`
- [x] Added any new workflows to `.github/workflows/workflow-testing.yml`
- [x] Added any new test data to `test_data`
- [x] When changing an existing module, have you bumped the version? Major.Minor.Patch - Major for breaking changes, Minor for feature updates and most things, Patch for bug fixes and small things like comment and variable name updates

## Notes
For the test suite it uses small subsets of databases. When `cmscan` or `hmmscan` are ran as part of a pipeline, the module is written to expect the user to provide database files, which additionally need to be indexed ahead of time.

### CODE REVIEW

**Comment by mjmansfi** [comment]
Hmm, something wrong with the `infernal` container. Investigating now.

File: configs/module-defaults.config [hunk 1]
```diff
@@ -206,6 +206,53 @@ params {
             publish_dir      = "last_dotplot"
             publish_results  = "all"
         }
+        'blast_makeblastdb' {
```
Comment (chris-cheshire) [inline configs/module-defaults.config:209]: Can you move these params to the blast PR? Ho do the tests pass without them?

File: tools/blast/main.nf [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+                      else filename }
+
+    container "quay.io/biocontainers/blast:2.9.0--pl526he19e7b1_7"
+
+    input:
+        val opts
+        tuple val(meta), path(fasta)
+
+    output:
+        tuple val(meta), path("*"), emit: blast_db
```
Comment (chris-cheshire) [inline tools/blast/main.nf:24]: should this be in blast PR?

File: tools/hmmer/main.nf [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    container "quay.io/biocontainers/hmmer:3.3--he1b5a44_1"
+
+    input:
+        val opts
+        tuple val(meta), path(fasta)
+
+    output:
+        tuple val(meta), path("*.hmmscan"), emit: hmmscan
+        tuple val(meta), path("*.tbl"), emit: table
```
Comment (chris-cheshire) [inline tools/hmmer/main.nf:25]: output as ext name pls

**Review by chris-cheshire — CHANGES_REQUESTED** [review]

**Review by chris-cheshire — APPROVED** [review]

**Pull request merged**